Author: Danny Milosavljevic <dannym@friendly-machines.com>
Date: 2026-05-31
Subject: Drop heap-status reporting from the bootstrap compiler
License: expat

The 3.2.0 compiler uses `TFPCHeapStatus` only for optional memory debug and status output.  The translated bootstrap does not need that diagnostic path, and keeping it would require extra runtime heap-status API surface.

Remove the reporting hooks.

--- a/compiler/cclasses.pas
+++ b/compiler/cclasses.pas
@@ -671,24 +671,15 @@
 
 
     procedure tmemdebug.start;
-
-      var
-        status : TFPCHeapStatus;
-
       begin
-        status:=GetFPCHeapStatus;
-        startmem:=status.CurrHeapUsed;
+        startmem:=0;
       end;
 
 
     procedure tmemdebug.stop;
-      var
-        status : TFPCHeapStatus;
       begin
         if startmem<>0 then
          begin
-           status:=GetFPCHeapStatus;
-           inc(TotalMem,startmem-status.CurrHeapUsed);
            startmem:=0;
          end;
       end;
@@ -703,11 +694,6 @@
 
     procedure tmemdebug.show;
       begin
-        write('memory [',infostr,'] ');
-        if TotalMem>0 then
-         writeln(DStr(TotalMem shr 10),' Kb released')
-        else
-         writeln(DStr((-TotalMem) shr 10),' Kb allocated');
       end;
 
 
--- a/compiler/comphook.pas
+++ b/compiler/comphook.pas
@@ -233,8 +233,6 @@
 ****************************************************************************}
 
 function def_status:boolean;
-var
-  hstatus : TFPCHeapStatus;
 begin
   def_status:=false; { never stop }
 { Status info?, Called every line }
@@ -244,9 +242,7 @@
         (status.currentline mod 100=0) then
        begin
          if status.currentline>0 then
-           Write(status.currentline,' ');
-         hstatus:=GetFPCHeapStatus;
-         WriteLn(DStr(hstatus.CurrHeapUsed shr 10),'/',DStr(hstatus.CurrHeapSize shr 10),' Kb Used');
+           WriteLn(status.currentline);
        end;
    end;
 {$ifdef macos}
--- a/compiler/compiler.pas
+++ b/compiler/compiler.pas
@@ -145,10 +140,6 @@
 uses
   aasmcpu;
 
-{$if defined(MEMDEBUG)}
-  {$define SHOWUSEDMEM}
-{$endif}
-
 var
   CompilerInitedAfterArgs,
   CompilerInited : boolean;
@@ -240,9 +227,6 @@
 var
   timestr    : string[20];
   linkstr    : string[64];
-{$ifdef SHOWUSEDMEM}
-  hstatus : TFPCHeapStatus;
-{$endif SHOWUSEDMEM}
   ExceptionMask : TFPUExceptionMask;
   totaltime : real;
 begin
@@ -391,11 +375,6 @@
         Raise;
       end;
   end;
-{$ifdef SHOWUSEDMEM}
-      hstatus:=GetFPCHeapStatus;
-      Writeln('Max Memory used/heapsize: ',DStr(hstatus.MaxHeapUsed shr 10),'/',DStr(hstatus.MaxHeapSize shr 10),' Kb');
-{$endif SHOWUSEDMEM}
-
   { Set the return value if an error has occurred }
   if status.errorcount=0 then
     result:=0
